weldx/measurement.py

```diff
@@ -207,25 +830,76 @@ def _signal_label(signal):
         )
 
         # draw edge labels
+        edge_labels = {
+            edge: _transformation_label(
+                edge[1], self._graph.edges[edge]["transformation"]
+            )
+            for edge in self._graph.edges
+        }
+
         draw_networkx_edge_labels(graph, positions, edge_labels, ax=axes)
 
         return axes
 
+    def to_tree(self) -> Dict:
+        """Get the content of the measurement chain as dictionary.
 
-@dataclass
-class Measurement:
-    """Simple dataclass implementation for generic measurements."""
+        Returns
+        -------
+        Dict:
+            Content of the measurement chain as dictionary.
 
-    name: str
-    data: Data
-    measurement_chain: MeasurementChain
+        """
+        return dict(
+            name=self._name,
+            data_source=self._source,
+            source_equipment=self._source_equipment,
+            source_signal=build_tree(self._graph, self._source.name),
+        )
+
+    @property
+    def source(self) -> SignalSource:
+        """Return the source of the measurement chain.
+
+        Returns
+        -------
+        SignalSource :
+            The source of the measurement chain
+
+        """
+        return self._source
+
+    @property
+    def source_name(self) -> str:
+        """Get the name of the source.
+
+        Returns
+        -------
+        str :
+            Name of the source
+
+        """
+        return self._source.name
+
+    @property
+    def transformation_names(self) -> List:
+        """Get the names of all transformations.
+
+        Returns
+        -------
+        List :
+            A list containing all transformation names
+
+        """
+        return [
+            self._graph.edges[edge]["transformation"].name for edge in self._graph.edges
+        ]
 
 
-# equipment ----------------------------------------------------------------------------
 @dataclass
-class GenericEquipment:
-    """Simple dataclass implementation for generic equipment."""
+class Measurement:
+    """Simple dataclass implementation for generic measurements."""
 
     name: str
-    sources: List = field(default_factory=lambda: [])
-    data_transformations: List = field(default_factory=lambda: [])
+    data: List["TimeSeries"]
```

Comment:
is this the list that links to the data that is also in the measurement chain?

Reply:
In theory, but there is no mechanism that enforces a synchronization. I am not sure if we still need this class. It doesn't provide any additional information.

Reply:
> I took a quick look over the new tutorial 👍
> 
> a few notes that would make it more accessible imo:
> 
> * plot the new measurement chain layout after each change
> * add an introduction explaining the concept (without code):
>   
>   * what is a signal?
>   * what is a transformation?
> * leave out analog/digital conversion from introduction as it is too complicated to get started imo
> * if possible I would prefer more code segments between the text and shorter text-segments (if it is supposed to focus on code)
> 
> also I would like to hear your thoughts @AndreasPittner @marscher **@rwinterk** (fake ping, need to get him on GH 😄 )

Rewrote most of the introduction trying to meet your requirements. Regarding the more code, less text bullet point: Please make a suggestion where you think this is necessary. Don't think the focus here is code alone. I tried to give an overall introduction.